R/slice.R

```diff
@@ -203,9 +203,9 @@ slice_max.data.frame <- function(.data, order_by, ..., n, prop, with_ties = TRUE
   }
 
   slice_impl(.data, {
-    n <- dplyr::n()
-    x <- vec_assert({{ order_by }}, size = n, arg = "order_by")
-    idx(x, n)
+    ...n <- dplyr::n()
+    x <- vec_assert({{ order_by }}, size = ...n, arg = "order_by")
+    idx(x, ...n)
```

Comment:
```suggestion
    order_by <- {{ order_by }}
    n <- dplyr::n()
    order_by <- vec_assert(order_by, size = n, arg = "order_by")
    idx(order_by, n)
```

Would it work if the first thing you did was evaluate the quosure? Before any local variables like `n` are defined?

Could do the same approach with `weight_by` in `slice_sample.data.frame()` I think

Reply:
In my mind this is kind of like calling `force(order_by)`

Reply:
Thanks. Thats' better.